107/src/main/java/org/ehcache/jsr107/ConfigurationMerger.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.jsr107;
+
+import org.ehcache.config.CacheConfiguration;
+import org.ehcache.config.CacheConfigurationBuilder;
+import org.ehcache.config.loaderwriter.DefaultCacheLoaderWriterConfiguration;
+import org.ehcache.config.xml.XmlConfiguration;
+import org.ehcache.internal.store.heap.service.OnHeapStoreServiceConfiguration;
+import org.ehcache.spi.loaderwriter.CacheLoaderWriter;
+import org.slf4j.Logger;
+
+import javax.cache.configuration.Configuration;
+
+import static org.ehcache.config.CacheConfigurationBuilder.newCacheConfigurationBuilder;
+
+/**
+ * ConfigurationMerger
+ */
+class ConfigurationMerger {
+
+  private final XmlConfiguration xmlConfiguration;
+  private final Jsr107Service jsr107Service;
+  private final Eh107CacheLoaderWriterFactory cacheLoaderWriterFactory;
+  private final Logger log;
+
+  ConfigurationMerger(org.ehcache.config.Configuration ehConfig, Jsr107Service jsr107Service, Eh107CacheLoaderWriterFactory cacheLoaderWriterFactory, Logger log) {
+    if (ehConfig instanceof XmlConfiguration) {
+      xmlConfiguration = (XmlConfiguration) ehConfig;
+    } else {
+      xmlConfiguration = null;
+    }
+    this.jsr107Service = jsr107Service;
+    this.cacheLoaderWriterFactory = cacheLoaderWriterFactory;
+    this.log = log;
+  }
+
+  <K, V> ConfigHolder<K, V> mergeConfigurations(String cacheName, Configuration<K, V> configuration) {
+    ConfigHolder<K, V> result = new ConfigHolder<K, V>();
+    CacheConfigurationBuilder<K, V> builder = newCacheConfigurationBuilder();
+    boolean useEhcacheExpiry = true;
```

Comment:
Can we move these declaration closer to their usage?  Putting them this far up makes me automatically assume the usage pattern is really complex (but it isn't).  Also I think we can avoid initing them when declaring, which means finding the 'true' condition doesn't require me to walk all the execution paths.